zebra-consensus/src/transaction/check.rs

```diff
@@ -105,3 +108,24 @@ pub fn output_cv_epk_not_small_order(output: &Output) -> Result<(), TransactionE
         Ok(())
     }
 }
+
+/// Check if a transaction is using the diabled sprout pool.
+///
+/// This check should be made only if the transaction block is above certain
+/// height where the sprout pool is disabled by consensus rules. This is after
+/// Canopy activation height.
```

Comment:
Code structure:

If we quote the consensus rule, and do the `Canopy` check inside the function:
- it will be easier for auditors to review the consensus rule, and
- it will be easier for developers to use the function correctly.

(And we won't need this comment telling callers to check Canopy.)

Reply:
Part of https://github.com/ZcashFoundation/zebra/pull/2399/commits/6c1de6d5809a2098101d6ec1c02b0e3ec935afbd